src/models/SearchResultsBib.ts

```diff
@@ -0,0 +1,81 @@
+import type { SearchResult } from "../types/searchTypes"
+import type { ElectronicResource } from "../types/bibTypes"
+
+export default class SearchResultsBib {
+  id: string
+  title: string
+  yearPublished?: string
+  materialType?: string
+  publicationStatement?: string
+  eResources?: ElectronicResource[]
+  numPhysicalItems: number
+
+  constructor(result: SearchResult) {
+    this.id = result["@id"] ? result["@id"].substring(4) : ""
+    this.title = this.getTitleFromResult(result)
+    this.yearPublished = this.getYearFromResult(result)
+    this.materialType = result.materialType[0]?.prefLabel || null
+    this.publicationStatement = result.publicationStatement?.length
+      ? result.publicationStatement[0]
+      : null
+    this.eResources = result.electronicResources || null
+    this.numPhysicalItems = result.numItemsTotal || 0
+  }
+
+  get url() {
+    return `/bib/${this.id}`
+  }
+
+  get numElectronicResources() {
+    return this.eResources?.length || 0
+  }
+
+  get hasPhysicalItems() {
+    return this.numPhysicalItems > 0
+  }
+
+  get numItems() {
+    return this.hasPhysicalItems
+      ? this.numPhysicalItems
+      : this.numElectronicResources
+  }
+
+  get resourceType() {
+    return this.hasPhysicalItems ? "Item" : "Resource"
+  }
+
+  get itemMessage() {
+    return `${this.numItems} ${this.resourceType}${
+      this.numItems !== 1 ? "s" : ""
+    }`
+  }
+
+  getTitleFromResult(result: SearchResult) {
+    if (!result.titleDisplay || !result.titleDisplay.length) {
+      const author =
+        result.creatorLiteral && result.creatorLiteral.length
+          ? ` / ${result.creatorLiteral[0]}`
+          : ""
+      return result.title && result.title.length
+        ? `${result.title[0]}${author}`
+        : ""
+    }
+    return result.titleDisplay[0]
+  }
+
+  getYearFromResult(result: SearchResult) {
+    const { dateStartYear, dateEndYear } = result
```

Comment:
This could use a confirmation with Sean/Brooke. I know that this is a tricky area that sometimes leads to weird results.